### PR TITLE
Bug-fix change owner address built-in function

### PR DESCRIPTION
**Pull request by miiu96**
- Extended the built-in function `ChangeOwnerAddress` in order to support an event when the function is called through a smart contract call cross-shard
- Added also an activation flag for this feature

### CODE REVIEW

File: builtInFunctions/changeOwnerAddress.go [hunk 1]
```diff
@@ -55,9 +65,26 @@ func (c *changeOwnerAddress) ProcessBuiltinFunction(
 		return nil, ErrNotEnoughGas
 	}
 	gasRemaining := computeGasRemaining(acntSnd, vmInput.GasProvided, c.gasCost)
+
+	vmOutput := &vmcommon.VMOutput{ReturnCode: vmcommon.Ok, GasRemaining: gasRemaining}
+	isCrossShardCallThroughASmartContract := check.IfNil(acntDst) && vmcommon.IsSmartContractAddress(vmInput.CallerAddr)
```
Comment (iulianpascalau) [inline builtInFunctions/changeOwnerAddress.go:70]: I think the if check and the call to addOutputTransferToVMOutput can be added after L85, in which case, the if condition simplifies by one check. You can even extract it in a new function to simplify how the code looks. L69 can remain, the code looks better with it

File: builtInFunctions/changeOwnerAddress.go [hunk 2]
```diff
@@ -55,9 +65,26 @@ func (c *changeOwnerAddress) ProcessBuiltinFunction(
...
+	if isCrossShardCallThroughASmartContract && c.enableEpochsHandler.IsChangeOwnerAddressCrossShardThroughSCEnabled() {
+		addOutputTransferToVMOutput(
```
Comment (iulianpascalau) [inline builtInFunctions/changeOwnerAddress.go:72]: unit test for this if?

File: builtInFunctions/changeOwnerAddress.go (hunk 1 above)
Comment (miiu96) [reply builtInFunctions/changeOwnerAddress.go:70]: refactored.

File: builtInFunctions/changeOwnerAddress.go (hunk 2 above)
Comment (miiu96) [reply builtInFunctions/changeOwnerAddress.go]: done

File: builtInFunctions/changeOwnerAddress.go [hunk 3]
```diff
@@ -81,6 +93,21 @@ func (c *changeOwnerAddress) ProcessBuiltinFunction(
 	return vmOutput, nil
 }
 
+func (c *changeOwnerAddress) addOutputTransferToVmOutputForCallThroughSC(acntDst vmcommon.UserAccountHandler, vmInput *vmcommon.ContractCallInput, vmOutput *vmcommon.VMOutput) {
+	isCrossShardCallThroughASmartContract := check.IfNil(acntDst) && vmcommon.IsSmartContractAddress(vmInput.CallerAddr)
```
Comment (sasurobert) [inline builtInFunctions/changeOwnerAddress.go:97]: It is easier to read if we have early exists.
if flagNotEnabled return
if NotCrossShardThroughSC return

Comment (miiu96) [reply builtInFunctions/changeOwnerAddress.go:97]: done

**Review by sasurobert — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**